src/tests/ftest/server/metadata_svc_ops.py

```diff
@@ -0,0 +1,141 @@
+            2. Create pool1 with specified property svc_ops_entry_age.
+            3. Create containers by ThreadManager.
+            4. Run specified metadata workload cycles in multiple test loops (N cycles per loop).
+            5. Create pool2 with property svc_ops_enable:0.
+            6. Create containers by ThreadManager on pool2.
+            7. To establish a "baseline" time (without duplicate rpc detection), perform test
+               step 4 on pool2, calculating average time per loop executed.
+            8. Compare all metadata workload times (with duplicate rpc detection) to the average
+               baseline time (without duplicate rpc).
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium,md_on_ssd
+        :avocado: tags=server,metadata
+        :avocado: tags=DuplicateRpcDetection,test_metadata_dup_rpc
+        """
+        num_of_cont = self.params.get("number_thread", '/run/metadata/*', default=1)
+        w_cycles = self.params.get("workload_test_cycles", '/run/metadata/*', default=5000)
+        t_loops = self.params.get("test_loops", '/run/metadata/*', default=10)
+        threshold_factor = self.params.get("threshold_factor", '/run/metadata/*', default=1.75)
+
+        self.log_step("Create pool with properties svc_ops_entry_age.")
+        pool1 = self.get_pool(dmg=self.dmg.copy())
+
+        self.log_step("Create containers by ThreadManager.")
+        container_manager = ThreadManager(
+            self.metadata_workload_test, self.get_remaining_time() - 30)
+        container_manager.add(
+            pool=pool1, num_of_cont=num_of_cont, workload_cycles=w_cycles, test_loops=t_loops)
+
+        self.log_step("Run specified metadata workload cycles in multiple test loops.")
+        results = container_manager.run()
+        num_failed = len(list(filter(lambda r: not r.passed, results)))
+        if num_failed > 0:
+            self.fail('#{} container create threads failed'.format(num_failed))
+
+        self.log_step("Create pool2 with property svc_ops_enable:0.")
+        params = {}
+        params['properties'] = "svc_ops_enabled:0"
+        self.add_pool(**params)
+
+        self.log_step("Create containers by ThreadManager on pool2.")
+        container_manager = ThreadManager(
+            self.metadata_workload_test, self.get_remaining_time() - 30)
+        container_manager.add(
+            pool=self.pool, num_of_cont=num_of_cont, workload_cycles=w_cycles, test_loops=t_loops)
+
+        self.log_step(
+            "To establish a baseline time without duplicate rpc detection, ",
+            "calculating average time per loop executed on pool2.")
+        base_results = container_manager.run()
+        base_average_time = statistics.mean(base_results[0].result)
+        num_failed = len(list(filter(lambda r: not r.passed, base_results)))
+        if num_failed > 0:
+            self.fail('#{} container create threads failed'.format(num_failed))
```

Comment:
Should we check for failure before getting the mean? In case the mean can't be calculated from some bad result?
```suggestion
        num_failed = len(list(filter(lambda r: not r.passed, base_results)))
        if num_failed > 0:
            self.fail('#{} container create threads failed'.format(num_failed))
        base_average_time = statistics.mean(base_results[0].result)
```